src/shims/unix/linux/eventfd.rs

```diff
@@ -28,16 +30,38 @@ struct Event {
     counter: u64,
     is_nonblock: bool,
     clock: VClock,
+    /// We have to store our own FdID in contrast to every other file descriptor out there, because
+    /// we are updating ourselves when writing and reading. Technically `Event` is like socketpair, but
+    /// it does not create two separate file descriptors. Thus we can't re-borrow ourselves via
+    /// `FileDescriptionRef::check_and_update_readiness` while already being mutably borrowed for read/write.
+    id: FdId,
 }
 
 impl FileDescription for Event {
     fn name(&self) -> &'static str {
         "event"
     }
 
+    fn get_epoll_ready_events<'tcx>(&self) -> InterpResult<'tcx, EpollReadyEvents> {
```

Comment:
```suggestion
    fn get_epoll_ready_events<'tcx>(&self) -> EpollReadyEvents {
```

Reply:
Interestingly after removing this, a strange error is thrown:
```bash
error[E0277]: the trait bound `shims::unix::linux::epoll::EpollReadyEvents: std::ops::FromResidual<std::ops::Yeet<rustc_const_eval::interpret::InterpError<'_>>>` is not satisfied
   --> src/shims/unix/fd.rs:101:9
    |
101 |         throw_unsup_format!("{}: epoll does not support this file description", self.name());
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::ops::FromResidual<std::ops::Yeet<rustc_const_eval::interpret::InterpError<'_>>>` is not implemented for `shims::unix::linux::epoll::EpollReadyEvents`
    |
note: required by a bound in `std::ops::try_trait::from_yeet`
   --> /home/byt/.rustup/toolchains/miri/lib/rustlib/src/rust/library/core/src/ops/try_trait.rs:343:8
    |
341 | pub fn from_yeet<T, Y>(yeeted: Y) -> T
    |        --------- required by a bound in this function
342 | where
343 |     T: FromResidual<Yeet<Y>>,
    |        ^^^^^^^^^^^^^^^^^^^^^ required by this bound in `from_yeet`
    = note: this error originates in the macro `throw_unsup_format` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I am not sure what's going on here yet.

Reply:
You need to remove the `?` at thr call sites

Reply:
Oh... I forgot about the default impl. You're absolutely correct